erc20-holdings/src/lib.rs

```diff
@@ -119,3 +125,43 @@ fn store_balance(transfers: erc20::TransferEvents, output: store::StoreAddBigInt
         }
     }
 }
+
+#[substreams::handlers::store]
+fn store_balance_usd(
+    transfers: erc20::TransferEvents,
+    balances: store::StoreGetBigInt,
+    prices: StoreGetProto<Erc20Price>,
+    output: store::StoreSetBigDecimal,
+) {
+    for transfer in transfers.items {
+        let token_price =
+            match prices.get_last(format!("chainlink_price:{}", &transfer.token_address)) {
+                Some(x) => BigDecimal::from_str(x.price_usd.as_str()).unwrap(),
+                None => BigDecimal::from_str("0").unwrap(),
+            };
+
+        let token_denominator = abi::erc20::functions::Decimals {}
```

Comment:
why not get it from `Erc20Price`? Contract calls are orders of magnitude slower and should be avoided at all costs